src/platform_impl/linux/wayland/event_loop.rs

```diff
@@ -695,10 +724,6 @@ impl<T> EventLoop<T> {
                         wid,
                     );
                 }
-                if refresh {
-                    unimplemented!()
-                    //sink.send_window_event(crate::event::WindowEvent::RedrawRequested, wid);
-                }
```

Comment:
The `refresh` needs to be handled here, because the `for_each` method will clear the flag after being called. This means that if the `refresh` flag had been set *before* `MainEventsCleared`, this will be lost.

I think the best way to handle this would be to remove the handling of `refresh` from this `for_each` method, and instead add to `WindowStore` a similar method specifically for handling the refreshes.